crates/trie/parallel/src/proof_task.rs

```diff
@@ -260,7 +554,78 @@ where
         (trie_cursor_factory, hashed_cursor_factory)
     }
 
+    /// Compute storage proof without consuming self.
+    ///
+    /// Borrows self immutably to allow transaction reuse across multiple calls.
+    /// Used by storage workers in the worker pool to avoid transaction creation
+    /// overhead on each proof computation.
+    #[inline]
+    fn compute_storage_proof(&self, input: StorageProofInput) -> StorageProofResult {
+        let (trie_cursor_factory, hashed_cursor_factory) = self.create_factories();
```

Comment:
this should be initialized once per storage proof worker, because it's always the same

Reply:
good note, thanks